src/ChannelServer/World/Entities/Components/Buffs.cs

```diff
@@ -0,0 +1,247 @@
+﻿// Copyright (c) Aura development team - Licensed under GNU GPL
+// For more information, see licence.txt in the main folder
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Melia.Channel.Network;
+using Melia.Channel.World.Entities;
+using Melia.Shared.Const;
+using Melia.Shared.EntityComponents;
+
+namespace Melia.Channel.World
+{
+	/// <summary>
+	/// Character buffs.
+	/// </summary>
+	public class Buffs : IUpdatableComponent
+	{
+		private readonly Dictionary<BuffId, Buff> _buffs = new Dictionary<BuffId, Buff>();
+
+		/// <summary>
+		/// The owner of this object.
+		/// </summary>
+		public ICombatEntity Owner { get; }
+
+		/// <summary>
+		/// Creates new instance for character.
+		/// </summary>
+		/// <param name="entity"></param>
+		public Buffs(ICombatEntity entity)
+		{
+			this.Owner = entity;
+		}
+
+		/// <summary>
+		/// Returns the amount of buffs in the collection.
+		/// </summary>
+		public int Count { get { lock (_buffs) return _buffs.Count; } }
+
+		/// <summary>
+		/// Adds given without updating the client. Replaces existing
+		/// buffs.
+		/// </summary>
+		/// <param name="buff"></param>
+		public void AddSilent(Buff buff)
+		{
+			lock (_buffs)
+				_buffs[buff.Id] = buff;
+		}
+
+		/// <summary>
+		/// Adds given buff and updates the client.
+		/// </summary>
+		/// <param name="buff"></param>
+		public void Add(Buff buff)
+		{
+			buff.IncreaseOverbuff();
+			this.AddSilent(buff);
+			var handler = ChannelServer.Instance.BuffHandlers.GetBuff(buff.Id);
```

Comment:
Shouldn't this happen in AddSilent as well? Otherwise the buff wouldn't get started there.

Reply:
I've moved the buff handlers to AddSilent/RemoveSilent since those are called within the counterpart functions which send the client update packets. Cuts down on the redundancy in my opinion.